### PR TITLE
Fix pvtIndex map for cases with non-active eqlnum regions.

### PR DESCRIPTION
**Pull request by totto82**

### CODE REVIEW

File: opm/core/simulator/initStateEquil.hpp [hunk 1]
```diff
@@ -371,13 +372,12 @@ namespace Opm
...
                 void setRegionPvtIdx(const Grid& G, const Opm::EclipseState& eclipseState, const RMap& reg) {
 
-                    regionPvtIdx_.reserve(reg.activeRegions().size());
                     std::vector<int> cellPvtRegionIdx;
                     extractPvtTableIndex(cellPvtRegionIdx, eclipseState, UgGridHelpers::numCells(G), UgGridHelpers::globalCell(G));
                     for (const auto& r : reg.activeRegions()) {
                         const auto& cells = reg.cells(r);
                         const int cell = *(cells.begin());
-                        regionPvtIdx_[r] = cellPvtRegionIdx[cell];
+                        regionPvtIdx_[reg.region(cell)] = cellPvtRegionIdx[cell];
```
Comment (bska) [inline opm/core/simulator/initStateEquil.hpp]: How is `reg.region(cell)` different from `r` in this case?

File: opm/core/simulator/initStateEquil.hpp [hunk 2]
```diff
@@ -263,6 +263,7 @@ namespace Opm
                     const auto& tables = eclipseState.getTableManager();
                     // Create (inverse) region mapping.
                     const RegionMapping<> eqlmap(equilnum(eclipseState, G));
+                    regionPvtIdx_.resize(rec.size());
```
Comment (atgeirr) [inline opm/core/simulator/initStateEquil.hpp]: Please resize the elements to something invalid, such as -1.

Comment (andlaus) [reply opm/core/simulator/initStateEquil.hpp]: do you mean s/resize/initialize/ ?

Comment (atgeirr) [reply opm/core/simulator/initStateEquil.hpp]: Yes I do...

```
const int invalidRegion = -1;
regionPvtIdx_.resize(rec.size(), invalidRegion);
```

File: opm/core/simulator/initStateEquil.hpp (hunk 1 above)
Comment (totto82) [reply opm/core/simulator/initStateEquil.hpp]: It doesn't. I miss read the region code.

File: opm/core/simulator/initStateEquil.hpp (hunk 2 above)
Comment (totto82) [reply opm/core/simulator/initStateEquil.hpp]: Fixed.

**Comment by totto82** [comment]
I have verified that the current version solves the problem. I am running a full test of the PR just to be sure that the refactoring of the init stuff doesn't have any other side effects.

**Comment by totto82** [comment]
I can also verify that all the big models are fine with this fix.

**Comment by atgeirr** [comment]
jenkins build this with downstreams please

**Pull request merged**